README.md
GEOS 3.4.2 backport to Ubuntu Precise, required for PostGIS 2 PPA: https://launchpad.net/~lwarx/+archive/postgis-pg93-bp/+packages
Use it together with PostgreSQL 9.3 installed from apt.postgresql.org


debian/changelog
geos (3.4.2-1~precise1) precise; urgency=low

  * New upstream release 3.4.2

 -- Max Arnold <arnold.maxim@gmail.com>  Mon, 18 Nov 2013 21:09:25 +0700

geos (3.3.8-3~precise1) precise; urgency=low

  * Misc tweaks

 -- Max Arnold <arnold.maxim@gmail.com>  Thu, 04 Jul 2013 02:27:22 +0700

geos (3.3.8-2~precise1) precise; urgency=high

  * Launchpad build for Precise

 -- Jerome Villeneuve Larouche <jlarouche@mapgears.com>  Thu, 25 Apr 2013 15:11:05 -0400

geos (3.3.8-1) unstable; urgency=low

  * New Upstream release 3.3.8

 -- Jerome Villeneuve Larouche <jlarouche@mapgears.com>  Thu, 25 Apr 2013 14:21:01 -0400

geos (3.3.3-1) unstable; urgency=low

  * New upstream bugfixing version. This is useful because correctly
    support new topology-enabled for Postgis 2.0.
  * Policy bumped to 3.9.3, no changes required.
  * Debhelper level set to 9.
  * Patch geos-config revised, partially merged upstream.

 -- Francesco Paolo Lovergine <frankie@debian.org>  Wed, 23 May 2012 15:30:33 +0200

geos (3.3.2-1) unstable; urgency=low

  * New upstream release.
  * Dropped geos-config patch, merged upstream.

 -- Francesco Paolo Lovergine <frankie@debian.org>  Sun, 26 Feb 2012 23:29:37 +0100

geos (3.3.1-1) unstable; urgency=low

  * New upstream major release.
    (closes: #648611)
  * Changed the C/C++ chain for development. See NEWS file for an explanation.
    Thanks strk for a useful discussion about that.
    (closes: #643658)
  * Patches refreshed and merged with upstream changes.
  * Added patch geos-config to fix library paths. This is coherent with
    upstream approach too.
  * Moved the C++ headers under libgeos++-dev automagically and added a break 
    against the old libgeos-dev all-in-one package.
  * The libgeos++-dev package depends on libgeos-dev for a couple of reasons:
      - a common header (geos/extern.h)
      - a program using C++ API could mixup C and C++ APIs so better being
        safe about that.
  * Merged NMU change in debian/rules to fix armhf FTBS.
    (closes: #641387)

 -- Francesco Paolo Lovergine <frankie@debian.org>  Wed, 22 Feb 2012 14:20:57 +0100

geos (3.2.2-3) unstable; urgency=low

  * New patch: swig. Fixed for strong version detection in ac_pkg_swig.m4.
    Merged from 3.3 series. Thanks strk. (closes: #631819)
  * Removed obsolete python build-deps, because now the suggested binding
    is via shapely. (closes: #616823)

 -- Francesco Paolo Lovergine <frankie@debian.org>  Mon, 18 Jul 2011 12:23:11 +0200

geos (3.2.2-2) unstable; urgency=low

  * Policy bumped to 3.9.2, no changes.
  * Fixed previous changelog entry.
  * Added CoordinateSequenceFactory.h patch to fix a missing declaration.
    (closes: #624949)
  * Added Bintree.cpp, Node.cpp, MonotoneChainBuilder.h, AbstractNode.cpp,
    and other patches to fix a missing cstddef header file inclusion in 
    some files and use of std namespace. This is required in current
    default gcc flavor.
  * Use -fpermessive at building time to work around some minor compatibility
    problems which prevent a successuful building.

 -- Francesco Paolo Lovergine <frankie@debian.org>  Wed, 25 May 2011 15:33:37 +0200

geos (3.2.2-1) unstable; urgency=low

  [ Alan Boudreault ]
  * New upstream release.
	
  [ Francesco Paolo Lovergine ]
  * Policy set to 3.9.1, without changes.
  * Merged log for post squeeze release.
  * Repository moved to git and changed Vcs-* fields as required.
  * Source format moved to 3.0 with quilt support.
  * Added ${misc:Depends} to bin pkgs due to dh use.
  * Added -dbg package for debugging symbols.
    (closes: #582273)

 -- Francesco Paolo Lovergine <frankie@debian.org>  Thu, 03 Mar 2011 13:40:04 +0100

geos (3.2.0-1) unstable; urgency=low

  * New upstream release.
  * Removed kobold among uploaders because no more active on DebianGIS packages.
  * Policy bumped to 3.8.3, without changes.
  * Changed a bit debian/rules to manage RC patch version.
  * Now, it does not install libtool helper file .la, as currently deprecated. 
  * Revised licenses around and updated the debian/copyright file.

 -- Francesco Paolo Lovergine <frankie@debian.org>  Thu, 21 Jan 2010 17:05:45 +0100

geos (3.1.1-1) unstable; urgency=low

  * New upstream release (bug fixes only).
  * Added me as an uploader.
  * Policy bumped to 3.8.2 (no changes).
  * Introduced a debian/control.in template to version automagically 
    the C++ binding package and dependencies.

 -- Francesco Paolo Lovergine <frankie@debian.org>  Fri, 26 Jun 2009 14:28:42 +0200

geos (3.1.0-1) unstable; urgency=low

  * New upstream release.
  * This version includes also a missing header inclusion, which caused FTBFS
    on armel. (Closes: #520447)

 -- Fabio Tranchitella <kobold@debian.org>  Fri, 27 Mar 2009 15:54:54 +0100

geos (3.0.3-2) unstable; urgency=low

  * debian/control: do not build the python bindings, they are deprecated by
    upstream and superseded by python-shapely.

 -- Fabio Tranchitella <kobold@debian.org>  Fri, 06 Mar 2009 13:28:29 +0100

geos (3.0.3-1) unstable; urgency=low

  [ Francesco Paolo Lovergine ]
  * New upstream release.
  * Updated debian/copyright adding copyright notice [lintian].
  * Policy bumped to 3.8.0 (no changes)
  * C++ version bumped in package name.
  * Updated homepage fields in debian/control to reflect moving into OSGeo
    infrastracture.
  * Watch file updated.
    (closes: #504470)
  * Patch gcc43.dpatch removed, merged upstream.
  * Moved to debhelper level 7. 
  * Some minor clean up in debian/rules.

  [ Fabio Tranchitella ]
  * debian/control: added libgeos-ruby1.8 and python-geos. (Closes: #509679)
  * debian/copyright: updated.

 -- Fabio Tranchitella <kobold@debian.org>  Thu, 05 Mar 2009 20:18:57 +0100

geos (3.0.0-5) unstable; urgency=low

  * debian/rules: fixed a FTBFS on arm eabi; patch from Riku Voipio.
    (Closes: #480345)

 -- Fabio Tranchitella <kobold@debian.org>  Mon, 12 May 2008 12:33:47 +0200

geos (3.0.0-4) unstable; urgency=low


debian/patches/Node.cpp
Index: geos/src/index/bintree/Node.cpp
===================================================================
--- geos.orig/src/index/bintree/Node.cpp	2011-11-30 16:54:52.000000000 +0100
+++ geos/src/index/bintree/Node.cpp	2011-12-02 14:36:08.000000000 +0100
@@ -16,6 +16,7 @@
 
 #include <cstddef>
 #include <cassert>
+#include <cstddef>
 
 #include <geos/index/bintree/Node.h>
 #include <geos/index/bintree/Key.h>
@@ -25,6 +26,8 @@
 namespace index { // geos.index
 namespace bintree { // geos.index.bintree
 
+using namespace std;
+
 Node*
 Node::createNode(Interval *itemInterval)
 {
Index: geos/src/geomgraph/Node.cpp
===================================================================
--- geos.orig/src/geomgraph/Node.cpp	2011-11-30 16:54:52.000000000 +0100
+++ geos/src/geomgraph/Node.cpp	2011-12-02 14:36:08.000000000 +0100
@@ -32,6 +32,7 @@
 #include <sstream>
 #include <vector>
 #include <algorithm>
+#include <cstddef>
 
 #ifndef GEOS_DEBUG
 #define GEOS_DEBUG 0
@@ -40,12 +41,13 @@
 #define COMPUTE_Z 1
 #endif
 
-using namespace std;
 using namespace geos::geom;
 
 namespace geos {
 namespace geomgraph { // geos.geomgraph
 
+using namespace std;
+
 /*public*/
 Node::Node(const Coordinate& newCoord, EdgeEndStar* newEdges)
 	:
Index: geos/src/index/quadtree/Node.cpp
===================================================================
--- geos.orig/src/index/quadtree/Node.cpp	2011-11-30 16:54:52.000000000 +0100
+++ geos/src/index/quadtree/Node.cpp	2011-12-02 14:36:08.000000000 +0100
@@ -25,6 +25,7 @@
 #include <string>
 #include <sstream>
 #include <cassert>
+#include <cstddef>
 
 #ifndef GEOS_DEBUG
 #define GEOS_DEBUG 0
@@ -41,6 +42,8 @@
 namespace index { // geos.index
 namespace quadtree { // geos.index.quadtree
 
+using namespace std;
+
 /* public static */
 std::auto_ptr<Node>
 Node::createNode(const Envelope& env)
